### ISSUE TITLE
LaTeXML hangs forever (100% cpu usage)

### ISSUE THREAD

**Issue by Grief** [brucemiller/LaTeXML#985]
Here is the document causing that behaviour:
```
\documentclass{mmaauth}
\usepackage{moreverb}
\usepackage[dvips,colorlinks,bookmarksopen,bookmarksnumbered,citecolor=red,urlcolor=red]{hyperref}



\begin{document}

\MOS{MSC 2010:
Primary 35R11;
Secondary 35B40, 44A10, 42B10}

\maketitle
\end{document}
```

You can find `mmauth.cls` here: http://arxmliv.kwarc.info/sty/mmaauth.cls

**Comment by dginev** [comment]
Thanks for the report! I've renamed the issue (and reproduced the infinite loop with `--includestyles`)

I am also impressed you found the sty directory of arxmliv!

As a clarification, I am labeling this issue as an "enhancement" as we are currently not claiming that latexml can natively interpret all class and style packages that native tex can. But that is certainly the goal, and this report is quite helpful, especially given the minimal example!

Will take a more detailed look soon.

**Comment by dginev** [comment]
A quick remark, given that the loop is in `\maketitle` - latexml does its own metadata processing, so even if the class file could be natively interpreted, the resulting XML will likely be quite flawed, as compared to adding a native binding. 

Certainly important to avoid the infinite loop though, so that needs a patch just for the principle of it.

**Comment by Grief** [comment]
Hi Deyan.
>I am also impressed you found the sty directory of arxmliv!
Right, just noticed that you are affiliated with this project. However, to my regret, that was not me. Even when I know where it is I cannot build a google search request to find it. That was my team leader who found this archive.

Anyway, thanks for the instant response, hope the test case I've provided is helpful in any way. Cheers!

**Comment by dginev** [comment]
Yes, the page you found contains work done just about 10 years ago, and has been largely discontinued since - everything of value was brought over to the main LaTeXML suite, exactly here :+1: 

That's why I mentioned that I was impressed you found that page. I am still periodically running latexml over the entire arXiv corpus, but any tangible results of that appear here as github issues / pull requests right away.

**Comment by dginev** [comment]
I took a closer look at the raw package interpretation prior to Bruce's upgrades.

The cause of the loop is subtle, as the loop itself is quite intricate and relies on a mixture of expansions and executions.

I could avoid the loop by adding the native definitions for `\pickup@font` and `\define@newfont` from latex.ltx, but that didn't make the document work cleanly - HTML was produced with 6 errors.

Given that, to my current understanding, LaTeXML's font machinery is on a rather higher level than latex's font internals, it does make sense to go the bounding route, as Bruce did, and avoid these concerns.

This does pose an open question about interpreting natively `.cls` files in the future, as I assume font work is common. Also, thanks for localizing the issue to `tracefnt.sty`, which was quite helpful - indeed all troubles are localized in there.
